### ISSUE TITLE
Tooltip too large and runs off-screen (cut off)

### ISSUE THREAD

**Issue by rgpublic** [Microsoft/TypeScript-Sublime-Plugin#638]
It should wrap properly instead of having to scroll horizontally. Making the font size smaller with "typescript_popup_font_size" doesnt seem to work either.

![image](https://user-images.githubusercontent.com/17726950/35056813-35841b10-fbb3-11e7-8c0f-b3b462a9d89d.png)

**Comment by stweedie** [comment]
This is also the case for error messages, making getting rid of them difficult as the 'x' to close is at the right side of the window, which might require scrolling the text editor window to close.

**Comment by stweedie** [comment]
turns out the fix is really simple

remove the last 'replace' call in escape_html of {package}\typescript\libs\text_helpers.py
should be:

```python
def escape_html(raw_string):
    """Escape html content

    Note: only use for short strings
    """
    return raw_string.replace('&', '&amp;').replace('<', '&lt;').replace('>', '&gt;').replace('\n', '<br>')
```

Sublime text will automatically wrap things, but the use of '&nbsp;' confuses their parsing I guess

**Comment by roblav96** [comment]
@stweedie I couldn't get your fix to work :/ Then again I have never python'ed before lol Maybe a PR if you get it working?

**Comment by jshado1** [comment]
@stweedie's solution didn't quite work for me (possibly a difference across OS?):

1. For the `text_helpers.py` fix, it's definitely an improvement (and the original code definitely seems wrong—why would you replace all space characters with non-breakable spaces??) and does cause auto-wrapping, but it leads to strange/arbitrary line-breaks:
![sublimetext3-typescript-popup-wrapping](https://user-images.githubusercontent.com/3012099/56083459-36534480-5e1d-11e9-8464-fad10fb4359f.png)

1. For the `quick_info.py` fix, I think `size` is not defined (as the popup does not appear at all when using it, and `size` does not appear anywhere else within the file). Instead, I just set the values to 400 and 800 respectively.

His solutions at least make it useable 🎉

**Comment by stweedie** [comment]
Yeah it's been awhile since I've looked into this issue. The changes to 'escape html' might be pretty far reaching, but they seemed to work in my setup. Granted, to really fix the issue I assume one would have to inspect the formatting of the response from the integrated tsc server and make changes according to that.

the 'size' variable was one I added myself, to get the size of the current viewport
